server/application-server/src/main/java/de/tum/cit/aet/helios/commit/github/GitHubPushMessageHandler.java

```diff
@@ -53,7 +53,17 @@ protected void handleInstalledRepositoryEvent(GHEventPayload.Push eventPayload)
       GHCommit commit = curRepo.getCommit(sha);
       commitSyncService.processCommit(commit, repository);
 
-      GHBranch branch = curRepo.getBranch(ref.split("/")[2]);
+      // Extract branch name by taking everything after the second slash
+      String[] refParts = ref.split("/");
```

Comment:
Cant we just use substring here?

Something like that (not tested): ref.substring(ref.indexOf("/") + 1);